lib/resque/scheduler/delaying_extensions.rb

```diff
@@ -257,55 +248,49 @@ def get_last_enqueued_at(job_name)
 
       private
 
+      def decode_without_nonce(job)
+        decode(job).delete_if {|k, _| k == 'nonce'}
+      end
+
       def job_to_hash(klass, args)
-        { class: klass.to_s, args: args, queue: queue_from_class(klass) }
+        job_to_hash_with_queue(queue_from_class(klass), klass, args)
       end
 
       def job_to_hash_with_queue(queue, klass, args)
         { class: klass.to_s, args: args, queue: queue }
       end
 
+      def encode_with_nonce(hash)
+        encode(hash.merge(nonce: rand))
+      end
+
       def remove_delayed_job(encoded_job)
         return 0 if Resque.inline?
 
         timestamps = redis.smembers("timestamps:#{encoded_job}")
 
         replies = redis.pipelined do
+          redis.zrem(:delayed_queue, [encoded_job])
```

Comment:
another nitpick: `zrem(:delayed_queue, encoded_job)` for consistency with other ZREM usages?

Reply:
We can't do that. The ruby implementation behaves different if given one item or an array of them.

Reply:
How does it behave different?

Reply:
If you pass it with an array it returns the count of removed items (0 / 1). If you pass it with an item directly it returns a true / false boolean.